src/nupic/engine/YAMLUtils.cpp

```diff
@@ -249,7 +249,17 @@ ValueMap toValueMap(const char *yamlstring,
       Value v = toValue(i->second, ps.dataType);
       if (v.isScalar() && ps.count != 1)
       {
-        NTA_THROW << "Parameter '" << key << "'. Bad value in runtime parameters. Expected array value but got scalar value";
+        if (key == "dim") {
+          // Spacial case for 'dim'; allow dimensions to be either an array {dim: [5]} 
+          // or a Scalar {dim: 5} which is treated as a single dimension array.
```

Comment:
OT, but do we need this dual functionality? Why not require just the [5] array case?

Reply:
> OT, but do we need this dual functionality? Why not require just the [5] array case?

You are probably right.  It will complicate the docs.  I did this because I kept forgetting the [ ]'s.  I will remove it.